### DISCUSSION TITLE
Decoding RawTransactionHex of SUBMIT_POST

### DISCUSSION THREAD

**Discussion by Mike00000001** [HPaulson/BitClout-API-Docs#12] [Q&A] [answered]
After converting from Hex the only plaintext RawTransactionHex seems to contain is information about an image/reclout/etc. The rest seems to be encoded somehow. Any idea how to decode the RawTransactionHex value?

**Comment by Mike00000001** [comment]
Here is an example of the RawTransactionHex that has readable data randomly pulled from the chain.

It is my assumption that data is encoded to (temporarily) hide the data on the chain until they are ready to release the method to decode. They could not hide image data because that must originate from the browser so the decode function would have to be released. Not sure about reclouts. My assumptions could be wrong.

Hex: 01304063fbd4e17cb5ac7912553216d005a48f029351d1ecd9ecc01d2c7ad749d300010214422731b0b6a1c972ecc7b7109f953d32b103c3ab7a0d0c050d0e4ea8566145aea4ec13052c00001c7b22426f6479223a22222c22496d61676555524c73223a6e756c6c7d00d46196afb6f0ced18ebc1600210214422731b0b6a1c972ecc7b7109f953d32b103c3ab7a0d0c050d0e4ea8566145020f497351756f7465645265636c6f75740100115265636c6f75746564506f737448617368208c620ea16e7dfe876dd0f4ab8c2abe2ec1d0e63b0a841ccb1c304611813030ac4730450221008570b854115f547bad8fa9832f6c34d140358cdadb5a459f7ee4ace331b3686e02204adeecc141d6db79cfa6de5355c51097a2d707260aeff90f6f3fb9b08a2b44b7

0@cûÔá|µ¬yU2Ð¤QÑìÙìÀ,z×IÓB'1°¶¡ÉrìÇ·=2±Ã«z

N¨VaE®¤ì,{"Body":"","ImageURLs":null}Ôa¯¶ðÎÑ¼!B'1°¶¡ÉrìÇ·=2±Ã«z

N¨VaEIsQuotedRecloutRecloutedPostHash b¡n}þmÐô«*¾.ÁÐæ;
Ë0F00¬G0E!p¸T_T{­©/l4Ñ@5ÚÛZE~ä¬ã1³hn JÞìÁAÖÛyÏ¦ÞSUÅ¢×&
ïùo?¹°+D·

(some chars not converting correctly on this site but you can see the clear text I mentioned relating to reclouts and images)

**Comment by Mike00000001** [comment]
Also, there is always a set of repeating data in the hex. In the above example:

0@cûÔá|µ¬yU2Ð¤QÑìÙìÀ,z×IÓB'1°¶¡ÉrìÇ·=2±Ã«z

N¨VaE®¤ì,{"Body":"","ImageURLs":null}Ôa¯¶ðÎÑ¼!B'1°¶¡ÉrìÇ·=2±Ã«z

N¨VaEIsQuotedRecloutRecloutedPostHash b¡n}þmÐô«*¾.ÁÐæ;
Ë0F00¬G0E!p¸T_T{­©/l4Ñ@5ÚÛZE~ä¬ã1³hn JÞìÁAÖÛyÏ¦ÞSUÅ¢×&
ïùo?¹°+D·

Duplicate entry: B'1°¶¡ÉrìÇ·=2±Ã«z

N¨VaE

Example 2:

2^_?ÝõVôQO¥ì§ôÁÂéÀ$2Ïd å¼\)y>P­j-s?µycåN]@`È |°
Ø§¤Ëñ	"D9iÐÛê£û$¤2oVw¤ñöajmêS0fc(!¼\)y>P­j-s?µycåN]@`È |°
F0D s¼¿_?"­	@áxR3[¤¾ç¹³u<ut­L8 [j`É¬Êâ
¾T§þ®uÔ$îån±Xµ¦

Duplicate entry: ¼\)y>P­j-s?µycåN]@`È |°

And this trend continues in each RawTransactionHex...

**Comment by HPaulson** [comment, marked as answer]
@2000c5 So far, nobody in the community (To my knowledge) has successfully decoded the hex. Here's what we know:

Transaction data such as the JSON string is easy to convert into plain text. It's currently believed that such JSON is freeform, thus engineers will, in the future, be able to have transactions on the chain with proprietary data. However, unlike the rest of the transaction data, the JSON string is inlined.

The rest of the hex is in an encoded packed-binary format, with an unknown pepper, making it extremely hard (Practically impossible) to decode without access to the pepper, which will come when node source is released in a few weeks from now.
